rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -245,10 +245,17 @@ object SystemProcesses {
         case isContractCall(produce, Seq(ack)) =>
           for {
             data <- blockData.get
-            _    <- produce(Seq(RhoType.Number(data.blockNumber), RhoType.Number(data.timeStamp)), ack)
+            _ <- produce(
```

Comment:
My only issue here is: would it be beneficial to make it a `SystemDeploy`? This would start to move us towards having only single mechanism of Scala <-> RhoLang interop (now we have like 3: `ConstructDeploy`, `SystemProcess` and `SystemDeploy`)

Reply:
@marcin-rzeznicki do you mean to define `SystemProcess`es part of `SystemDeploy`? How this will work with non-blockchain Rholang?

Reply:
@marcin-rzeznicki system processes are called from Rholang. How would you call a system deploy from Rholang?

Reply:
I'm not sure if I'm understanding it correctly, so correct me if I'm wrong. But, at least in this case, the system process simply binds `Ref` to a channel.  I think it can be perfectly simulated in RhoLang. I see similar (?) examples in PoS eg. `state.set("allBonds",state.get("allBonds").set(userPk,amount))` so I'm concluding it can be made into a SystemDeploy

Reply:
Now I see what you mean. Since we can't move all system processes to system deploys, I'm not sure if it is worth a refactoring.

Reply:
Ok let's leave it for now